src/openms/include/OpenMS/METADATA/SpectrumMetaDataLookup.h

```diff
@@ -290,6 +290,24 @@ namespace OpenMS
       bool override_spectra_references = false, 
       std::vector<ProteinIdentification> proteins = std::vector<ProteinIdentification>());
 
+
+    /**
+     * @brief Adds missing ion mobility information to peptide identifications.
+     * 
+     * This function adds missing ion mobility (IM) information to the peptide identifications in the given vector.
+     * The missing IM information is retrieved from the specified file.
+     * 
+     * @param peptides The vector of peptide identifications to update.
+     * @param filename The name of the file containing the missing IM information.
```

Comment:
While I am not a huge fan of passing filenames we already do something similar for:
    static bool addMissingRTsToPeptideIDs(std::vector<PeptideIdentification>& peptides, const String &filename,
      bool stop_on_error = false);
      
So maybe we just need to be a bit more explicit about the file formats behind the filename.
```
       @param filename Name of a raw data file (e.g. mzML) for looking up RTs
```

Reply:
e.g. one downside is that if we want to reannotate RT and IM we would need to parse the file two times